src/contracts/permissions/PermissionController.sol

```diff
@@ -0,0 +1,176 @@
+        address account
+    ) {
+        require(isAdmin(account, msg.sender), NotAdmin());
+        _;
+    }
+
+    /**
+     *
+     *                         INITIALIZING FUNCTIONS
+     *
+     */
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize() external initializer {}
+
+    /**
+     *
+     *                         EXTERNAL FUNCTIONS
+     *
+     */
+
+    /// @inheritdoc IPermissionController
+    function setAdmin(address account, address admin) external onlyAdmin(account) {
+        // Add the admin to the account's admins
+        // If the admin is already set, the set will fail
+        EnumerableSet.AddressSet storage admins = _permissions[account].admins;
+        require(admins.add(admin), AdminAlreadySet());
+
+        emit AdminSet(account, admin);
+    }
+
+    function removeAdmin(address account, address admin) external onlyAdmin(account) {
+        EnumerableSet.AddressSet storage admins = _permissions[account].admins;
+
+        // Remove the admin from the account's admins
+        // If the admin is not set, the remove will fail
+        require(admins.remove(admin), AdminNotSet());
+
+        emit AdminRemoved(account, admin);
+    }
+
+    /// @inheritdoc IPermissionController
+    function setDelegate(
+        address account,
+        address delegate,
+        address target,
+        bytes4 selector
+    ) external onlyAdmin(account) {
+        AccountPermissions storage permissions = _permissions[account];
+
+        bytes32 targetSelector = _encodeTargetSelector(target, selector);
+        require(!permissions.delegatePermissions[delegate].contains(targetSelector), DelegateAlreadySet());
+
+        // Add the delegate to the account's permissions
+        permissions.delegatePermissions[delegate].add(targetSelector);
+        permissions.permissionDelegates[targetSelector].add(delegate);
+
+        emit DelegateSet(account, delegate, target, selector);
+    }
+
+    /// @inheritdoc IPermissionController
+    function removeDelegate(
+        address account,
+        address delegate,
+        address target,
+        bytes4 selector
+    ) external onlyAdmin(account) {
+        AccountPermissions storage permissions = _permissions[account];
+
+        bytes32 targetSelector = _encodeTargetSelector(target, selector);
+        require(permissions.delegatePermissions[delegate].contains(targetSelector), DelegateNotSet());
+
+        // Remove the delegate from the account's permissions
+        permissions.delegatePermissions[delegate].remove(targetSelector);
+        permissions.permissionDelegates[targetSelector].remove(delegate);
+
+        emit DelegateRemoved(account, delegate, target, selector);
+    }
+
+    /**
+     *
+     *                         INTERNAL FUNCTIONS
+     *
+     */
+
+    /// @dev Encodes the target and selector into a single bytes32 value
+    function _encodeTargetSelector(address target, bytes4 selector) internal pure returns (bytes32) {
```

Comment:
Would be good to have reversability unit test. Where you get the same starting target,selector values.

_encodeTargetSelector(target, selector) => bytes32 targetSelector
_decodeTargetSelector(targetSelector) => (target, selector)

Reply:
Added basic reversibility: https://github.com/Layr-Labs/eigenlayer-contracts/pull/870/commits/343b44e11fe3b3824941fbb6fe05a19307b93c25